### DISCUSSION TITLE
Is it possible to add a class to html tag in server-side?

### DISCUSSION THREAD

**Discussion by hadikman** [vercel/next.js#67747] [Help]
### Summary

Hi
I want to add 'no-scrollbar' class to html tag only when the user is at root page, and for other pages, it must shows scrollbar.
I did this in client side rendering but the problem is when the page loads the user sees the scrollbar and the class is applied after the page is fully loaded.
I want to don't show scrollbar for when the page loads.

The code used for client side rendering:

```
export default function UpdateHtmlAtClient() {
  const pathname = usePathname()
  const isRootPath = pathname === '/'

  React.useEffect(() => {
    const tagEl = document.getElementsByTagName('html')
    if (isRootPath) {
      tagEl[0].classList.add('no-scrollbar')
    } else {
      tagEl[0].classList.remove('no-scrollbar')
    }
  }, [isRootPath])

  return null
}
```

Thank you

### Additional information

```js
Version: Nextjs 14.2.3
Folder Structure: App Router
```

**Comment by icyJoseph** [comment]
You might want to experiment with `useLayoutEffect`, and if you see complaints on the server side console, then you can look into using, https://usehooks-ts.com/react-hook/use-isomorphic-layout-effect - look at the bottom of the page.

**Reply by hadikman** [reply]
I used useIsomorphicLayoutEffect from @react-spring/web but didn't solve the issue. It still shows the scrollbar for first time and then disappear when the class added.

**Comment by hadikman** [comment]
Is it okay to use Pages Router in App Router and use _document and add/remove a class base on current path?
